src/rpc/blockchain.cpp

```diff
@@ -2668,21 +2668,42 @@ UniValue CreateUTXOSnapshot(
 
     afile << metadata;
 
+    std::map<uint256, std::vector<std::pair<uint32_t, Coin>>> mapCoins;
+    unsigned int iter{0};
     COutPoint key;
+    uint256 last_hash;
     Coin coin;
-    unsigned int iter{0};
+    std::vector<std::pair<uint32_t, Coin>> coins;
+
+    auto write_coins_to_file = [&](AutoFile& afile, const uint256& last_hash, const std::vector<std::pair<uint32_t, Coin>>& coins) {
+        afile << last_hash;
+        afile << static_cast<uint16_t>(coins.size());
```

Comment:
4e194640a74ff784212e615c8e88c375cdfed039: In `primitives/transaction.h` we serialize std::`vector<CTxOut> vout` with a simple `s << tx.vout;` without any reference to the size of the vector. Not sure if any magic is happening there under the hood that I missed.

Also, `uint16_t` implies a limit of 65,536 outputs per transaction. I don't think that's a consensus rule?

Reply:
A transaction with 65,537 OP_RETURN outputs should fit in a block.

If I start with P2TR outputs with this calculator, that's 2,818,159 vbyte. https://bitcoinops.org/en/tools/calc-size/

And then subtract 32 bytes per output: 2,818,159 - 65537 * 32 = 720,975 vbyte

cc @murchandamus can you add OP_RETURN to the dropdown? :-)

In any case it seems unsafe to rely on the block size here.

Reply:
Hm, but OP_RETURNs are not included in the UTXO set and we are serializing the UTXO set here, so I think this could still not happen like this. But I think you are right there are non-standard cases imaginable that make this possible, like just sending to OP_TRUE for example. So we should still make this robust.

Anyway, I am using CompactSize now in #29612 :)